src/api/helpers/send-email-confirm-email-address/index.test.ts

```diff
@@ -32,22 +33,70 @@ describe('helpers/send-email-confirm-email-address', () => {
     sendEmail.confirmEmailAddress = sendEmailConfirmEmailAddressSpy;
   });
 
-  test('it should call sendEmail.confirmEmailAddress and return success=true', async () => {
-    const result = await confirmEmailAddressEmail.send(context, mockUrlOrigin, account.id);
+  describe('when verificationHash and verificationExpiry do not exist', () => {
+    test('it should call sendEmail.confirmEmailAddress and return success=true', async () => {
+      const result = await confirmEmailAddressEmail.send(context, mockUrlOrigin, account.id);
+
+      const { email, verificationHash } = account;
+
+      const name = getFullNameString(account);
 
-    const { email, verificationHash } = account;
+      expect(sendEmailConfirmEmailAddressSpy).toHaveBeenCalledTimes(1);
+      expect(sendEmailConfirmEmailAddressSpy).toHaveBeenCalledWith(email, mockUrlOrigin, name, verificationHash, account.id);
 
-    const name = getFullNameString(account);
+      const expected = {
+        success: true,
+        emailRecipient: mockAccount.email,
+      };
 
-    expect(sendEmailConfirmEmailAddressSpy).toHaveBeenCalledTimes(1);
-    expect(sendEmailConfirmEmailAddressSpy).toHaveBeenCalledWith(email, mockUrlOrigin, name, verificationHash, account.id);
+      expect(result).toEqual(expected);
+    });
+  });
 
-    const expected = {
-      success: true,
-      emailRecipient: mockAccount.email,
-    };
+  describe('when verificationHash exists and verificationExpiry has elapsed', () => {
+    test('it should call sendEmail.confirmEmailAddress and return success=true and not use account verificationHash', async () => {
+      await accounts.update(context, account.id, { verificationExpiry: DATE_24_HOURS_IN_THE_PAST(), verificationHash: 'mock-hash' });
+      const updatedAccount = await accounts.get(context, account.id);
+      const result = await confirmEmailAddressEmail.send(context, mockUrlOrigin, updatedAccount.id);
 
-    expect(result).toEqual(expected);
+      const { email, verificationHash } = updatedAccount;
+
+      const name = getFullNameString(account);
+
+      expect(sendEmailConfirmEmailAddressSpy).toHaveBeenCalledTimes(1);
+      // should generate new hash so not called with account hash
+      expect(sendEmailConfirmEmailAddressSpy).not.toHaveBeenCalledWith(email, mockUrlOrigin, name, verificationHash, account.id);
+
```

Comment:
Can we add a space between line 58/59, 66/67?

Reply:
Seems that the test description is wrong and should have "NOT call `sendEmail.confirmEmailAddress`

Reply:
it does call sendEmail.confirmEmailAddress but doesn't use the account's verification hash

Reply:
have updated description